File: samples/inventory-service/inventory-py/inventory/item.py
from __future__ import annotations

from dataclasses import dataclass


@dataclass
class AddItemInput:
    """Details of a new item to add to the catalog."""

    sku: str
    name: str
    initial_quantity: int


@dataclass
class AddItemOutput:
    item_id: str
    created_at: str

File: samples/inventory-service-with-predefined-models/inventory-go/types/item.go
package types

// AddItemInput describes a new item to add to the catalog.
type AddItemInput struct {
	SKU             string `json:"sku"`
	Name            string `json:"name"`
	InitialQuantity int    `json:"initialQuantity"`
}

// AddItemOutput describes the creation of the item.
type AddItemOutput struct {
	ItemID    string `json:"itemId"`
	CreatedAt string `json:"createdAt"`
}
